sfcsample/VDecAccelVA.cpp

```diff
@@ -0,0 +1,743 @@
+                               sizeof(m_vaProcBuffer),
+                               1,
+                               (uint8_t*)&m_vaProcBuffer,
+                               &vaBufferID );
+    assert(VASUCCEEDED(vaStatus));
+    if (VASUCCEEDED(vaStatus))
+                vaBufferIDs.push_back(vaBufferID);
+
+
+    if (vaBufferIDs.size())
+    {
+        vaStatus = vaRenderPicture(
+            m_vaDisplay,
+            m_vaContextID,
+            &(vaBufferIDs.at(0)),
+            vaBufferIDs.size());
+    }
+
+    //va end picture
+    vaStatus = vaEndPicture(m_vaDisplay, vaContextID);
+    if (VAFAILED(vaStatus))
+        printf("vaEndPicture fail.");
+
+
+    auto gfx_surface_buf = lock_surface(m_sfcIDs[0], false);
+    if (gfx_surface_buf == nullptr)
+    {
+        printf("Fail to lock gfx surface\n");
+        return 1;
+    }
+
+    //write to yuv file
+    FILE* sfc_stream = fopen("sfc_sample.yuv", "wb");
+    uint32_t file_size = m_DecodeDesc.sfc_widht * m_DecodeDesc.sfc_height * 4; //ARGB format
+    fwrite(gfx_surface_buf, file_size, 1, sfc_stream);
+
+    fclose(sfc_stream);
+
+    unlock_surface(m_sfcIDs[0]);
+    Close();
+}
+
+// check vaQueryVideoProcPipelineCaps
+int mvaccel::VDecAccelVAImpl::check_process_pipeline_caps(DecodeDesc& desc)
+{
+    VAProcPipelineCaps caps;
+
+    VAProcColorStandardType inputCST[VAProcColorStandardCount];
+    VAProcColorStandardType outputCST[VAProcColorStandardCount];
+    caps.input_color_standards = inputCST;
+    caps.output_color_standards = outputCST;
+
+    m_in4CC.resize(vaMaxNumImageFormats(m_vaDisplay));
+    m_out4CC.resize(vaMaxNumImageFormats(m_vaDisplay));
+    caps.input_pixel_format = &m_in4CC.at(0);
+    caps.output_pixel_format = &m_out4CC.at(0);
+
+    VABufferID filterIDs[VAProcFilterCount];
+    uint32_t filterCount = 0;
+
+    VAStatus vaStatus = VA_STATUS_SUCCESS;
+    vaStatus = vaQueryVideoProcPipelineCaps(
+        m_vaDisplay,
+        m_vaContextID,
+        filterIDs,
+        filterCount,
+        &caps
+        );
+    if (VAFAILED(vaStatus))
+    {
+        printf("vaQueryVideoProcPipelineCaps fail\n");
+        return 1;
+    }
+
+    m_in4CC.resize(caps.num_input_pixel_formats);
+    m_out4CC.resize(caps.num_output_pixel_formats);
+
+    return 0;
+}
+
+int mvaccel::VDecAccelVAImpl::create_resources()
+{
+    if (m_sfcIDs.empty())
+        m_sfcIDs.resize(1);
+
+    // Create addition surfaces for scaled video output
+    uint16_t width = m_DecodeDesc.sfc_widht;
+    uint16_t height = m_DecodeDesc.sfc_height;
+
+    //prepare sfc surface attribs
+    DecodeDesc SfcDesc;
+    SfcDesc.sfcformat = MVFOURCC('A','R','G','B');
+    VASurfaceAttribArray Sfc_vaSurfAttribs;
+    prepare_surface_attribs(SfcDesc, Sfc_vaSurfAttribs, true);
+
+    uint32_t surfaceType = m_surfaceType;
+    m_surfaceType = (uint32_t)SfcDesc.sfcformat;
+
+    for(uint32_t i = 0; i < m_sfcIDs.size(); i++)
+    {
+        create_surface(width, height, Sfc_vaSurfAttribs, m_sfcIDs[i]);
```

Comment:
If  no special purpose,  could you call vaCreateSurfaces directly?

Reply:
replace function create_surface with vaCreateSurfaces